systemd/systemd.go

```diff
@@ -358,6 +358,25 @@ type Systemd interface {
 	// threads if enabled, etc) part of the unit, which can be a service or a
 	// slice.
 	CurrentTasksCount(unit string) (uint64, error)
+	// Run a command
+	Run(command []string, opts *RunOptions) ([]byte, error)
+}
+
+// KeyringMode describes how the kernel keyring is setup, see systemd.exec(5)
+type KeyringMode string
+
+const (
+	KeyringModeInherit KeyringMode = "inherit"
+	KeyringModePrivate KeyringMode = "private"
+	KeyringModeShared  KeyringMode = "shared"
+)
+
+// RunOptions can be passed to systemd.Run()
+type RunOptions struct {
+	// XXX: alternative we could just have `Propertes []string` here
```

Comment:
Based on the usage in `cmd_helper.go`, I think this will eventually be necessary if this is to be used there. Having a field for each property doesn't seem appropriate given the possible values for the other properties.
```suggestion
	// XXX: alternatively we could just have `Properties []string` here
```